File: GO/DecodeRoman.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package GO

import "strings"

/**
 * Receive a roman number and decode to a integer value.
 */
func Decode(roman string) int {
	var result int = 0
	roman = strings.ToUpper(roman)

	for i := 0; i < len(roman)-1; i++ {
		if singleDecode(roman[i]) < singleDecode(roman[i+1]) {
			result -= singleDecode(roman[i])
		} else {
			result += singleDecode(roman[i])
		}
	}
	result += singleDecode(roman[len(roman)-1])

	return result
}

func singleDecode(letter byte) int {
	var result int = 0

	switch letter {
	case 'M':
		result = 1000
	case 'D':
		result = 500
	case 'C':
		result = 100
	case 'L':
		result = 50
	case 'X':
		result = 10
	case 'V':
		result = 5
	case 'I':
		result = 1
	}

	return result
}

File: PHP/DecodeRoman.php
<?php
/**
 * @version PHP
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
namespace PHP;

//echo "The result is "+Decode( "XX" );

class DecodeRoman {

    /**
     * Receive a roman number and decode to a integer value.
     */
    function Decode( $roman ) : int {
        $result = 0;
        $roman = strtoupper( $roman );
    
        for ( $i = 0; $i < strlen( $roman )-1; $i++ ) {
            if ( $this->singleDecode( substr( $roman, $i, 1 ) ) < $this->singleDecode( substr( $roman, $i+1, 1 ) ) ) {
                $result -= $this->singleDecode( substr( $roman, $i, 1 ) );
            } else {
                $result += $this->singleDecode( substr( $roman, $i, 1 ) );
            }
        }
        $result += $this->singleDecode(substr( $roman, strlen( $roman )-1, 1 ));
    
        return $result;
    }

    function singleDecode( $letter ) : int {
        $result = 0;
    
        switch ($letter) {
        case 'M':
            $result = 1000;
            break;
        case 'D':
            $result = 500;
            break;
        case 'C':
            $result = 100;
            break;
        case 'L':
            $result = 50;
            break;
        case 'X':
            $result = 10;
            break;
        case 'V':
            $result = 5;
            break;
        case 'I':
            $result = 1;
            break;
        }
    
        return $result;
    }

}
